projects/project2/com.test.firstscalaprogram/src/com/test/firstscalaprogram/MainObject.scala

```diff
@@ -0,0 +1,206 @@
+package com.test.disdupProject
+import org.apache.spark._
+import org.apache.spark.rdd.RDD
+import scala.collection.mutable.HashMap
+import scala.collection.mutable.ListBuffer
+import scala.xml._
+import shapeless.ops.tuple.Length
+
+
+object MainObject {
+  
+  val pathOfDataFile : String = "E:\\tmp\\Spark_Workspace\\CommaSeperatedData.txt"
+  val appName : String = "Disdup"
+  val outputDirectory : String = "E:\\tmp\\Spark_Workspace\\output"
+  
+  
+ def main(args:Array[String]) =
+ { 
+   try
+   {  
+     NaiveDistinct()
+     //BlockBased()
+     //partitionEqually()
+   }
+   catch
+   {
+     case 
+     e : Exception => println(e)
+     e.printStackTrace()
+   }
+ }
+ 
+ 
+ def NaiveDistinct()
+ {
+   try
+   {
+    val conf:SparkConf = new SparkConf().setAppName("Disdup").setMaster("local[*]")
+    val sc = SparkContext.getOrCreate(conf)
+    
+    val startTime = System.currentTimeMillis();
+    val textFile = sc.textFile("E:\\tmp\\Spark_Workspace\\CommaSeperatedData.txt", 2)
+    
+    ////////////////////Naive Distinct Function////////////////
+    
+    val testing = textFile.flatMap(record => 
+      {
+        record.split("\n")  
+      }
+      ).distinct()
+    testing.saveAsTextFile("E:\\tmp\\Spark_Workspace\\output")
+    
+    ////////////////////////////////////////////////////////
+    
+    val endTime = System.currentTimeMillis();
+    val processingTime = endTime - startTime
+    println("Time taken: "+processingTime)
+   
+   }
+   catch
+   {
+     case 
+     e : Exception => println(e)
+     e.printStackTrace()
+   }
+ }
+ 
+ 
+ def partitionEqually()
+ {
+   try
+   {
+    val conf:SparkConf = new SparkConf().setAppName(appName).setMaster("local[*]")
+    val sc = SparkContext.getOrCreate(conf)
+    
+    val startTime = System.currentTimeMillis();
+    val textFile = sc.textFile(pathOfDataFile)
+    
+    val wordCounts = getCounts(textFile)
+    
+    val totalPartitions = 2
```

Comment:
Is this supposed to be static, or should get all the possible (available) cores/nodes?

Reply:
No, they should be dynamic. 
Should I get the 

1. The number of workers is the number of executors minus one sc.getExecutorStorageStatus.length - 1.
or
2. The number of cores per worker can be obtained by executing java.lang.Runtime.getRuntime.availableProcessors on a worker.

Reply:
Because we are running this code on a single machine, so I was asking if I should get the number of executors or the number of cores in my system?

Reply:
Sure, you should assume that the code should run elsewhere (local mode, cluster mode) so there shouldn't be such restriction.

Reply:
Got it. Thank you.